lib/ace/ext/papeeria/spellchecker.coffee

```diff
@@ -1,39 +1,60 @@
 define( ->
-  # This will be removed and replaced with real data from server.
-  # BTW, this is kinda template for data to send from server.
-  TEST_JSON_TYPOS = {
-    "blablabla": [
-      "blah-blah-blah"
-      "blabber"
-      "blabbed"
-      "salable"
-    ]
-  }
 
+  class Spellchecker
+    constructor: (@editor) ->
+      @typosUrl = null          # url used to fetch typos
+      @suggestionsUrl = null    # url used to fetch corrections for a word
+      @typosHash = null         # hash used to check whether the typos list has been changed
+      @language = null          # language code, e.g. `en_US`
+
+    _fetchTypos: (hash) =>
+      $.getJSON(@typosUrl, null, (typosArray) =>
```

Comment:
Pls add `.fail` callback, so that  we could report errors somewhere (e.g. in the spellchecker dropdown). Like this:
```
constructor: (@editor, @errorCounter) ->
...
_fetchTypos: (hash) =>
  $.getJSON()
      .fail(=> @errorCounter.inc())
```